### DISCUSSION TITLE
Issues with new release

### DISCUSSION THREAD

**Discussion by macumber** [BuildingComponentLibrary/bcl-manifest#19] [General]
Hey everyone!  I posted a new test release today, https://github.com/automaticmagic/automaticmagic-measures/releases,  just to make sure things were still working on the BCL.  I saw three webhooks fire off from my repo to the BCL (created, published, and released).  The created and published ones were rejected by the BCL but the released one got a 200 success.  However, the BCL page for my org https://bcl.nrel.gov/repos/e7d3dccc-04a8-11ec-a8db-129880bd8b7d shows that there was an error fetching https://api.github.com/repos/automaticmagic/automaticmagic-measures/zipball/v0.1.2.  That link works in a browser for me.  Is there some order dependency between the three webhooks?  Does BCL retry if the zipball fetch fails?

![image](https://user-images.githubusercontent.com/1327850/173254279-d4eac573-c5eb-4c52-9fe0-22ab25fe3f0d.png)

I tried to redeliver the released webhook and got this error reply:

`<html><head><title>Request Rejected</title></head><body>An error has occurred. Please consult with your administrator.<br><br>Your support ID is: 16321039148529138914<br><br><a href='javascript:history.back();'>[Go Back]</a></body></html>`

**Comment by rd2** [comment]
**INITIAL ISSUE DESCRIPTION**: _I'm also looking into a similar issue as @macumber. Trying to get TBD released in BCL as **v2.4.1**. I get a **404** error for the **"action" : "released"** webhook/payload (Dan's successful - I'm not) with the following (similar) error message:_

`<html><head><title>Request Rejected</title></head><body>An error has occurred. Please consult with your administrator.<br><br>Your support ID is: 16321039148541232921<br><br><a href='javascript:history.back();'>[Go Back]</a></body></html>`

_This could be a completely separate issue, but the feedback is the same._

---

**EDIT** : Just deleted v2.4.1, moved on to [v2.4.2](https://github.com/rd2/tbd/releases/tag/v2.4.2). Only change on the repo side (i.e. v2.4.1 vs v2.4.2) was adding the **.md** extension to the root LICENSE. As with @macumber, GitHub now reports success ("**200**"), yet getting the same feedback from BCL.

<img width="567" alt="v242" src="https://user-images.githubusercontent.com/40636643/173256886-27b8d682-b913-4675-bd2b-c77720d6237e.png">

---

Tried again with [v2.4.3](https://github.com/rd2/tbd/releases), same result.

<img width="567" alt="v243" src="https://user-images.githubusercontent.com/40636643/173373695-5605b3e5-4549-4a50-b387-57397a684f0e.png">

**Comment by kflemin** [comment]
@macumber and @rd2 Hi!  This was an error on the BCL side (expired token). Sorry for the hassle. I've updated and manually added your releases to the BCL. Everything should be good now, but let me know if you run into any more issues. thanks! -Katherine

**Reply by rd2** [reply]
Thanks @kflemin, glad it's working! If I download from BCL however, I get v2.4.1 (and not the most recent v2.4.3). This may be due to my repeated attempts with successive patched versions. If there's no easy fix on your side, should I simply re release? Or move on to v2.4.4.?

<img width="567" alt="v241" src="https://user-images.githubusercontent.com/40636643/173439222-845f51b7-3509-4383-985e-b8caa66a7a4a.png">

**Reply by kflemin** [reply]
@rd2, I just downloaded from the page in your screenshot... the version modified in the measure.xml file that I downloaded has date 2022-06-12...is that not the latest version?

Also, I can see v2.4.3 from this search: https://bcl.nrel.gov/api/search/thermal%20bridging

Let me know if it still looks wrong for you...

**Reply by rd2** [reply]
Thanks for the follow-up @kflemin. My efforts in trying to get BCL to accept the latest release included re-releasing v2.4.1, then releasing v2.4.2, then 2.4.3 (each with slight changes). Some of these attempts occurred on 2022-06-12, the latest (v2.4.3) on 2022-06-13 (success, see figure).

<img width="567" alt="v243" src="https://user-images.githubusercontent.com/40636643/173620480-32a5e8b0-6148-48bb-980b-4ce72b842d0e.png">

Yet if one downloads TBD from BCL, it's v2.4.1 (one can CTRL-F "v2.4" in the _ua.md_ file under resources). The latest [release](https://github.com/rd2/tbd/releases) is v2.4.3. Any webhook payload on your side with "v2.4.1" or "v2.4.2" can be discarded, as I have deleted them from TBD's releases. Depending on what you dig up, I can always release v2.4.4 if needed.

**Comment by kflemin** [comment]
@rd2, I removed and re-added the v2.4.3 release with the same results...The 'webhook' and my manual process pulls the source code . zip from this page: https://github.com/rd2/tbd/releases/tag/v2.4.3.  That one has version 2.4.1 in the ua.rb file...which is what i'm seeing on BCL as well. Perhaps something went wonky in the release? If you want to try making a new one go for it, it should work this time.

**Comment by rd2** [comment]
<img width="567" alt="tags" src="https://user-images.githubusercontent.com/40636643/173775389-cd54a29b-623a-4fcc-a41a-6843b8cd3dd8.png">

Once again, thanks for your efforts @kflemin. Took me a while to figure out what was happening. In the above figure (about an hour ago), there's a 48-hour lag between the .zip/.tar.gz files I locally created/dragged/dropped vs what GitHub generates from the current source code (last committed/pushed 2 days ago) at the time of the release. A GitHub user wouldn't necessarily notice this gap if they dragged/dropped + released all under a minute. So I guess BCL instructions are pulling the .zip content that GitHub prepares behind the scenes at release (not the .zip file I prepared). So I just committed/pushed a recent change to master (a minor, yet relevant update). GitHub's .zip file and the one I dragged/dropped should now be the same.

<img width="567" alt="rerelease" src="https://user-images.githubusercontent.com/40636643/173777749-b2e8a364-18b9-4fbe-bef8-7faf8aac43d3.png">

The issue I seem to have now is syncing with BCL.

<img width="567" alt="404" src="https://user-images.githubusercontent.com/40636643/173777971-dad0fb6a-9a89-4a36-b64e-d7c4492cd35d.png">

I suspect BCL doesn't allow overwriting an existing "v2.4.3" tag? If so, I think BCL measure publishers should be allowed to edit releases through GitHub webhooks (GitHub allows editing a release). Could I therefore ask you again to manually reset BCL's v2.4.3 TBD release? The alternative, it seems, would be to release a v2.4.4 (which I would like to avoid, if I can). Merci !

<img width="567" alt="bcl" src="https://user-images.githubusercontent.com/40636643/173778340-6aa8eff7-b509-4ddf-b58b-f64fcf5760ea.png">

**Comment by kflemin** [comment]
@rd2 Okay, I just deleted v2.4.3 from BCL (you are right that it will not overwrite an existing tag on BCL).  Fingers crossed that it works this time! :)

**Reply by rd2** [reply]
Works !! Thanks a bunch @kflemin !!

**Reply by rd2** [reply]
As this is a discussion (not an issue), how does NREL feel with maybe revising which types of webhook requests from GitHub should be admissible?

- should subsequent re-releases (with the same tag) be allowed? If abused, it's ~unethical (but GitHub allows it - people make mistakes)
- should developers be able to even delete a BCL-distributed release?
- etc.

**Reply by kflemin** [reply]
These are good questions!  At the least, it might be good for BCL to allow re-releases...the BCL should probably match what users would see on the individual GitHub repos, and if that's allowed on github we should probably handle it on BCL as well.   I'll keep it in mind!

**Comment by brgix** [comment]
@kflemin , @macumber (Dan's co-author, so I'm dragging him into this) : I again have a similar issue with TBD v3.0.3 (last night). The recent deliveries look like this:

<img width="567" alt="deliver" src="https://user-images.githubusercontent.com/98104985/197184462-d6e7ac61-0672-4f4a-9363-4573aaeca69a.png">

I think we can ignore the failed re-release (I believe re-releases always fail ?!). But the final ("200" : success) message looked promising:

`{"message": "Released added. Indexing in progress...check the BCL dashboard for results"}`

Alas, TBD v3.0.2 remains the latest available version on BCL. TBD v3+ were all (successfully) released/published in the same manner (same settings). Yesterday's released is the only hiccup. Anything on BCL's side? Thanks.

**Reply by kflemin** [reply]
hi @brgix! hi @macumber!
this was caused by expired credentials on my end, sorry about that! I went ahead and manually added your latest release and everything looks good: https://bcl.nrel.gov/repos/6721d9ea-0704-11ec-810a-4aec1d5be697

just for reference, the BCL uses the "published" delivery to add a new release, so if that one fails it won't get indexed into the bcl. but again, the error was on my end.

**Reply by brgix** [reply]
@kflemin fantastic! Merci!

**Reply by macumber** [reply]
Thanks Kat!

**Comment by brgix** [comment]
@kflemin , @macumber : Just released v3.1, and am getting similar issues:
<img width="567" alt="v310" src="https://user-images.githubusercontent.com/98104985/205691835-8736337a-0366-498d-b76c-6cc8cfcc8a25.png">
Anything popping up on BCL's side, as before? Or something on my side?

**Reply by kflemin** [reply]
@brgix, this was again a credentials issue on the BCL side, ugh!  BCL didn't have permissions to download the release. I've fixed it manually and uploaded 3.1.0 to the BCL.  Hopefully next time it will work! sorry about that!

**Reply by brgix** [reply]
Thanks @kflemin !

**Comment by brgix** [comment]
@kflemin , @macumber : Just released v3.1.1 yesterday:
<img width="567" alt="v311" src="https://user-images.githubusercontent.com/98104985/207895581-74629e7a-87e7-42dc-a80b-b725cf6040ea.png">
... not popping up on BCL. Still wondering if it's something on my side (vs BCL)? Let me know if I need to tweak a parameter or two. Thanks.

**Reply by kflemin** [reply]
I did another manual update, it should be on BCL now!

**Reply by brgix** [reply]
Merci!

**Comment by brgix** [comment]
Hey @kflemin : Yup, that time of the year. Just released TBD v3.3.0, and got another **400 error** on `release.published` : 

`{"detail":"Bad Request. Missing 'action' key set to 'released' and/or 'release' key in payload"}`

Can you take a look on your side? Thanks.
Denis (+ @macumber)

**Comment by kflemin** [comment]
Ack, sorry! Fixed and released on BCL now: https://bcl.nrel.gov/repos/6721d9ea-0704-11ec-810a-4aec1d5be697

**Reply by brgix** [reply]
Merci @kflemin !

**Comment by brgix** [comment]
Bonjour @kflemin, @macumber. Maybe related to recent BCL difficulties, but the latest TBD [v3.4.1](https://github.com/rd2/tbd/releases/tag/v3.4.1) has a 404 failed delivery error. It's certainly not popping up under BCL. Let me know if you need details. Merci!

**Reply by kflemin** [reply]
sorry! the release should be on BCL now!

**Reply by brgix** [reply]
Merci @kflemin - all good!

**Comment by brgix** [comment]
Hi @kflemin ... yep, another 404 failed delivery error ([v3.4.2](https://github.com/rd2/tbd/releases/tag/v3.4.2)). Let me know if I need to do anything more on my end. Thanks.

**Reply by kflemin** [reply]
Hmm, this time I don't know why it didn't work. Usually it's because the token expired but not this time.  Would you mind sending me the error message you can see on your end?  

I did add the release locally so you should be all set!

**Comment by brgix** [comment]
Here's the "release.created", "release.released" & "release.published" feedback (from what I can tell, exactly the same; not very informative):

<html><head><title>Request Rejected</title></head><body>An error has occurred. Please consult with your administrator.<br><br>Your support ID is: 1779917883621227788<br><br><a href='javascript:history.back();'>[Go Back]</a></body></html>

_... let me know if you want me to share the actual (automated) requests._

**Reply by kflemin** [reply]
this is perfect, thank you. any change there's a timestamp that you can see?
